### PR TITLE
Adding description about Audit and Fault Pipeline into v6 documentation

### PR DESCRIPTION
**Pull request by WojcikMike**
Adding description about audit and fault pipeline. 

Related to this pull request:
https://github.com/Particular/NServiceBus/pull/3240

### CODE REVIEW

File: nservicebus/pipeline/customizing-v6.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ The following lists describe some of the common stages which you can build your
 
 ### Outgoing Pipeline Stages
 * Operation specific processing: There is a dedicated stage for each bus operation (e.g. Send, Publish, Subscribe, ...). Behaviors can use one of the following contexts: `IOutgoingSendContext`, `IOutgoingPublishContext`, `IOutgoingReplyContext`, `ISubscribeContext`, `IUnsubscribeContext`.
+* Audit message processing: In here behaviors have access to audited message and audit address. Behaviors should use `IAuditContext` to enlist in this stage.
```
Comment (andreasohlund) [inline nservicebus/pipeline/customizing-v6.md]: "to the audited message" ?

File: nservicebus/pipeline/customizing-v6.md [hunk 2]
```diff
@@ -50,6 +50,8 @@ The following lists describe some of the common stages which you can build your
...
+* Fault message processing: This is a dedicated stage for processing faulty messages. Behaviors have access to message, exception and error address. This stage provides `IFaultContext` to it's behaviors.
```
Comment (andreasohlund) [inline nservicebus/pipeline/customizing-v6.md]: Think we need to clarify that this is "messages being moved to the error queue" ?

File: nservicebus/pipeline/customizing-v6.md (hunk 1 above)
Comment (WojcikMike) [inline nservicebus/pipeline/customizing-v6.md]: IAuditContext has property message with a comment: The message to be audited. Do your question refer to:
a. The message is not audted yet. 
b. That this is actually the audit message, not the original message that my comment is suggesting?

File: nservicebus/pipeline/customizing-v6.md (hunk 2 above)
Comment (WojcikMike) [inline nservicebus/pipeline/customizing-v6.md]: What about:
"This is a dedicated stage for processing faulty messages that are being moved to the error queue" ?

Comment (andreasohlund) [inline nservicebus/pipeline/customizing-v6.md]: :+1:

File: nservicebus/pipeline/customizing-v6.md (hunk 1 above)
Comment (andreasohlund) [inline nservicebus/pipeline/customizing-v6.md]: I was just missing a `the` in the sentence :)

I guess to be correct its "access to the message to be audited/sent to the audit queue"

Comment (WojcikMike) [inline nservicebus/pipeline/customizing-v6.md]: :) missed the 'the'.

**Comment by WojcikMike** [comment]
This PR should be merged with the actual change:
https://github.com/Particular/NServiceBus/pull/3240

**Comment by SimonCropp** [comment]
this needs a rebase

**Comment by andreasohlund** [comment]
Core pull is now merged so this one can now be completed

**Pull request merged**